src/Doctrine/Orm/Filter/AbstractUuidFilter.php

```diff
@@ -0,0 +1,200 @@
+            [$alias, $field, $associations] = $this->addJoinsForNestedProperty($property, $alias, $queryBuilder, $queryNameGenerator, $resourceClass, Join::INNER_JOIN);
+        }
+
+        $metadata = $this->getNestedMetadata($resourceClass, $associations);
+
+        if ($metadata->hasField($field)) {
+            $values = $this->convertValuesToTheDatabaseRepresentationIfNecessary($queryBuilder, $this->getDoctrineFieldType($property, $resourceClass), $values);
+            $this->addWhere($queryBuilder, $queryNameGenerator, $alias, $field, $values);
+
+            return;
+        }
+
+        // metadata doesn't have the field, nor an association on the field
+        if (!$metadata->hasAssociation($field)) {
+            return;
+        }
+
+        // association, let's fetch the entity (or reference to it) if we can so we can make sure we get its orm id
+        $associationResourceClass = $metadata->getAssociationTargetClass($field);
+        $associationMetadata = $this->getClassMetadata($associationResourceClass);
+        $associationFieldIdentifier = $associationMetadata->getIdentifierFieldNames()[0];
+        $doctrineTypeField = $this->getDoctrineFieldType($associationFieldIdentifier, $associationResourceClass);
+
+        $associationAlias = $alias;
+        $associationField = $field;
+
+        if ($metadata->isCollectionValuedAssociation($associationField) || $metadata->isAssociationInverseSide($field)) {
+            $associationAlias = QueryBuilderHelper::addJoinOnce($queryBuilder, $queryNameGenerator, $alias, $associationField);
+            $associationField = $associationFieldIdentifier;
+        }
+
+        $values = $this->convertValuesToTheDatabaseRepresentationIfNecessary($queryBuilder, $doctrineTypeField, $values);
+        $this->addWhere($queryBuilder, $queryNameGenerator, $associationAlias, $associationField, $values);
+    }
+
+    /**
+     * Converts values to their database representation.
+     */
+    private function convertValuesToTheDatabaseRepresentationIfNecessary(QueryBuilder $queryBuilder, ?string $doctrineFieldType, array $values): array
+    {
+        if ($doctrineFieldType && Type::hasType($doctrineFieldType)) {
+            $doctrineType = Type::getType($doctrineFieldType);
+            $platform = $queryBuilder->getEntityManager()->getConnection()->getDatabasePlatform();
+            $databaseValues = [];
+
+            foreach ($values as $value) {
+                try {
+                    $databaseValues[] = $doctrineType->convertToDatabaseValue($value, $platform);
+                } catch (ConversionException $e) {
+                    $this->logger->notice('Invalid value conversion value to its database representation', [
+                        'exception' => $e,
+                    ]);
+                    $databaseValues[] = null;
+                }
+            }
+
+            return $databaseValues;
+        }
+
+        return $values;
```

Comment:
we should not need this this is done by Doctrine when applying parameters no? If this is really needed could you tell me why? Also you could have a guard clause in the callee and reduce the function complexity.